python/lsst/daf/butler/datastores/s3Datastore.py

```diff
@@ -208,9 +213,10 @@ def _write_in_memory_to_artifact(self, inMemoryDataset: Any, ref: DatasetRef) ->
         # _toBytes is not implemented
         try:
             serializedDataset = formatter.toBytes(inMemoryDataset)
+            log.debug(f"Writing file directly to: '{location.uri}''")
             self.client.put_object(Bucket=location.netloc, Key=location.relativeToPathRoot,
                                    Body=serializedDataset)
-            log.debug("Wrote file directly to %s", location.uri)
+            log.debug(f"Successfully wrote file directly to {location.uri}")
         except NotImplementedError:
             with tempfile.NamedTemporaryFile(suffix=location.getExtension()) as tmpFile:
                 formatter._fileDescriptor.location = Location(*os.path.split(tmpFile.name))
```

Comment:
I think we also need a log message around line 225 (new) where we upload this temporary file. This is the code path that will trigger for fits files.

Reply:
Adding line 225 (new), before it attempts the put_object via a temporary file.